.github/workflows/haskell.yml

```diff
@@ -197,8 +197,8 @@ jobs:
     - name: Regenerate hie.yaml and confirm that it is in sync
       run: ./scripts/gen-hie.sh
 
-  nix-build:
-    name: Build nix derivations and enter nix-shell
+  nix-cache-check:
+    name: Check nix required derivations are cached (built by hydra or cicero)
```

Comment:
I don't understand this bit :thinking: 
At the moment, because our main CI workflow is building and running via cabal (rather than nix), we have this job that is run nightly, checking that everything still builds with nix. 
What does it mean to check that the derivations are cached?  Are they being built and cached if not already cached, and not ran if cached?   Wouldn't it make more sense, nightly, to build them without cache?

Reply:
Now we have both cicero and hydra runing on PR and master branch so the idea was to check that there are indeed building master branch correctly. But now I understand that the aim was more to have easy access to build logs in case of nix shell failing. So I'll remove that commit, and use `nix develop` as suggested in your initial comment.